### PR TITLE
Feature/210 npm dependency updates

### PR DESCRIPTION
**Pull request by nublikasa**
Initial reviews for current code

### CODE REVIEW

File: src/components/util/Rivet.tsx [hunk 1]
```diff
@@ -149,9 +149,9 @@ const parseRivetHidden = (hide?: Hidden) => ({
  */
 export type ComponentDecorator<T extends Props> = ( props: T ) => string;
 
-export const rivetize = <T extends React.HTMLAttributes<HTMLElement>>(Component: React.ComponentType<T>): React.SFC<T & Props> =>
+export const rivetize = <T extends React.HTMLAttributes<HTMLElement>>(Component: React.ComponentType<any>): React.FC<any & Props> =>
```
Comment (nublikasa) [inline src/components/util/Rivet.tsx]: 👀 This line is particularly complex and the types aren't happy. The only way for me to make it work so far is to changes the types to any.

File: src/components/Input/Input.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import * as React from 'react';
 import * as Rivet from '../util/Rivet';
 import { renderInput, TextProps } from './common';
 
-const Input : React.SFC<TextProps & React.InputHTMLAttributes<HTMLInputElement>> = 
+const Input : any & React.FC<TextProps & React.InputHTMLAttributes<HTMLInputElement>> =
```
Comment (nublikasa) [inline src/components/Input/Input.tsx]: 👀 This is another hack where 'any' type is being used

File: src/components/Input/Select.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import * as React from 'react';
 import * as Rivet from '../util/Rivet';
 import { renderInput, TextProps } from './common';
 
-const Select : React.SFC<TextProps & React.SelectHTMLAttributes<HTMLSelectElement>> = 
+const Select : React.FC<any & TextProps & React.SelectHTMLAttributes<HTMLSelectElement>> =
```
Comment (nublikasa) [inline src/components/Input/Select.tsx]: 👀 This is another hack where 'any' type is being used

File: src/components/Input/Textarea.tsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ import * as React from 'react';
 import * as Rivet from '../util/Rivet';
 import { renderInput, TextProps } from './common';
 
-const Textarea :  React.SFC<TextProps & React.TextareaHTMLAttributes<HTMLTextAreaElement>> = 
+const Textarea : React.FC<any & TextProps & React.TextareaHTMLAttributes<HTMLTextAreaElement>> =
```
Comment (nublikasa) [inline src/components/Input/Textarea.tsx]: 👀 This is another hack where 'any' type is being used

File: src/components/List/List.tsx [hunk 5]
```diff
@@ -26,7 +26,7 @@ const asListItem = (child : React.ReactChild) =>
     : child;
 
 /** Ensure all children are <li> elements. */
-const asListItems = (children?: React.ReactNode) => 
+const asListItems = (children?: any & React.ReactNode) =>
```
Comment (nublikasa) [inline src/components/List/List.tsx]: 👀 This is another hack where 'any' type is being used

File: src/components/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -24,14 +24,14 @@ interface PaginationItemProps {
...
-const Pagination : React.SFC<PaginationProps & React.HTMLAttributes<HTMLDivElement>> =
+const Pagination : React.FC<PaginationProps & React.HTMLAttributes<HTMLDivElement>> =
  ({ align, children, className, size, ...attrs }) => {
     const classes = classNames({
       ['rvt-pagination']: true,
       [`rvt-pagination--${align}`]: align,
       [`rvt-pagination--${size}`]: size
     });
-    const wrappedChildren = React.Children.map(children, (child: React.ReactElement<any>) => {
+    const wrappedChildren = React.Children.map(children, (child: any & React.ReactElement<any>) => {
```
Comment (nublikasa) [inline src/components/Pagination/Pagination.tsx]: 👀 This is another hack where 'any' type is being used

File: src/setupTests.ts [hunk 7]
```diff
@@ -1,15 +1,17 @@
-/*
-Copyright (C) 2018 The Trustees of Indiana University
-SPDX-License-Identifier: BSD-3-Clause
-*/
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
+import '@testing-library/jest-dom/extend-expect';
```
Comment (athill) [inline src/setupTests.ts]: 🆒

**Comment by athill** [comment]
Do you know why you had to change the functional component types from `SFC` to `FC`? Is that just a change to `@types/react`? Just curious.

**Comment by nublikasa** [comment]
> Do you know why you had to change the functional component types from `SFC` to `FC`? Is that just a change to `@types/react`? Just curious.

SFC has been deprecated. Even if you use SFC, the library simply use the FC alias instead internally.

**Comment by athill** [comment]
Got it. Thanks!

**Pull request closed without merging**